src/Integration/LocalServices/ISourceControlledFileSystem.cs

```diff
@@ -30,31 +30,40 @@ namespace SonarLint.VisualStudio.Integration
     internal interface ISourceControlledFileSystem : ILocalService
     {
         /// <summary>
-        /// Queues a write operation for a file path. New or edit is determined internally at the time of this method execution.
+        /// Queues a write operation for given file paths. New or edit is evaluated at the time of this method execution.
         /// </summary>
-        /// <param name="filePath">File path that want to write to</param>
-        /// <param name="fileWriteOperation">The actual write operation, return false if failed</param>
-        void QueueFileWrite(string filePath, Func<bool> fileWriteOperation);
+        void QueueFileWrites(IEnumerable<string> filePaths, Func<bool> fileWriteOperation);
 
         /// <summary>
-        /// Queues a write operation for given file paths. New or edit is determined internally at the time of this method execution.
+        /// Returns whether all given files exist on disk or in a queue to be written. <seealso cref="QueueFileWrites(IEnumerable{string}, Func{bool})"/>
         /// </summary>
-        void QueueFileWrites(IList<string> filePaths, Func<bool> fileWriteOperation);
+        bool FilesExistOrQueuedToBeWritten(IEnumerable<string> filePaths);
 
         /// <summary>
-        /// Returns whether the file exists on disk or in a queue to be written. <seealso cref="QueueFileWrite(string, Func{bool})"/>
+        /// Checks out the file if the solution is under source control, and then executes all the queued file writes.  <seealso cref="QueueFileWrite(string, Func{bool})"/>
```

Comment:
The `cref` should be to `cref="QueueFileWrites(IEnumerable{string}, Func{bool})"` (although I'm not sure they are really much use in practice).